blog/2024-03-15-ROS-integration/index.md

```diff
@@ -0,0 +1,280 @@
+
+MotionJPEG (MJPEG) is a common format for video compression and is often used for video streaming. This format compresses each frame of video as a separate JPEG image, which can be useful for easily storing and processing images.
+
+If your camera supports MJPEG, you can set the `pixel_format` parameter to `raw_mjpeg` to publish JPEG images Here's an example of how to do this with a config file `usb_cam_params.yaml`:
+
+```yaml
+usb_cam:
+  ros__parameters:
+    video_device: "/dev/video0"
+    image_width: 640
+    image_height: 480
+    pixel_format: "raw_mjpeg"
+```
+
+We can run the `usb_cam` node to start capturing images and publishing them to the `/image_raw` topic using the following command:
+
+```bash
+ros2 run usb_cam usb_cam_node_exe --ros-args --params-file ./usb_cam_params.yaml
+```
+
+Once the `usb_cam` node is running, we can start the custom ROS 2 node that listens to the `/image_raw` topic and stores the images in ReductStore.
+
+```bash
+ros2 run reduct_camera capture_and_store
+```
+
+The `capture_and_store` node will start listening to the `/image_raw` topic and store the images in ReductStore as they are received.
+
+
+### Use ReductStore CLI to inspect stored images
+To get started, you need to install the ReductStore CLI by following the instructions [**here**](<https://cli.reduct.store/en/latest/>).
+
+Once installed, you can create an alias for the URL of your ReductStore instance using the `rcli alias add` command. For example:
+
+```bash
+rcli alias add -L  https://play.reduct.store play
+```
+
+You can then use the `rcli` command to inspect the stored images. For instance, to export the image data from the `ros-bucket` bucket to a local directory, you can use the following command:
+
+```bash
+rcli export folder --ext jpeg play/ros-bucket ./exported-data
+```
+
+This command exports all images from the `ros-bucket` bucket to the `./exported-data` folder with the JPEG file extension.
+
+## Best practices
+
+There are several best practices to consider when integrating ReductStore. Here are a few to keep in mind:
+
+- Use non-blocking operations to avoid blocking the main thread of the ROS 2 node. This ensures that the node can continue processing other tasks while waiting for data to be stored.
+- Serialize data before storing it in ReductStore in a cross-platform binary format to ensure compatibility with different systems and programming languages.
+- Create a ReductStore bucket with a [**FIFO**](<https://www.reduct.store/docs/how-does-it-work#bucket>) quota to prevent disk overwriting in the future.
+- Use token authentication to protect your data. You can generate an access token using either the [**Web Console**](<https://github.com/reductstore/web-console>) or the [**CLI client**](<https://cli.reduct.store/>).
+- Use the [**Web Console**](<https://github.com/reductstore/web-console>) or [**ReductCLI**](<http://cli.reduct.store/>) for data replication or backup purposes.
```

Comment:
use the web console or reductcli to set up data replication between two instances
use reductcli for manual replication or backup